src/http/conf.rs

```diff
@@ -31,3 +31,35 @@ pub unsafe fn ngx_http_conf_get_module_loc_conf(
     let http_conf_ctx = (*cf).ctx as *mut ngx_http_conf_ctx_t;
     *(*http_conf_ctx).loc_conf.add(module.ctx_index) as *mut ngx_http_core_loc_conf_t
 }
+
+/// # Safety
+///
+/// The caller has provided a value `ngx_http_upstream_srv_conf_t. If the `us` argument is null, a
+/// None Option is returned; however, if the `us` internal fields are invalid or the module index
+/// is out of bounds failures may still occur.
+pub unsafe fn ngx_http_conf_upstream_srv_conf_immutable<T>(
+    us: *const ngx_http_upstream_srv_conf_t,
+    module: &ngx_module_t,
+) -> Option<*const T> {
+    if us.is_null() {
+        return None;
+    }
+    let cf: *const T = (*us).srv_conf.offset(module.ctx_index as isize) as *const T;
+    Some(cf)
+}
+
+/// # Safety
+///
+/// The caller has provided a value `ngx_http_upstream_srv_conf_t. If the `us` argument is null, a
+/// None Option is returned; however, if the `us` internal fields are invalid or the module index
+/// is out of bounds failures may still occur.
+pub unsafe fn ngx_http_conf_upstream_srv_conf_mutable<T>(
```

Comment:
based on this: https://rust-lang.github.io/api-guidelines/naming.html then it may be renamed:
`ngx_http_conf_upstream_srv_conf_as_mut`